pkg/urbit/vere/dawn.c

```diff
@@ -426,22 +426,44 @@ u3_dawn_vent(u3_noun seed)
     u3z(oct); u3z(fut);
   }
 
-  {
-    u3l_log("boot: retrieving sponsor keys\r\n");
+  pon = u3_nul;
+  while (c3__czar != rank) {
+    //  retrieve +point:azimuth of pos (sponsor of ship)
+    //
+    u3_noun who = u3dc("scot", 'p', u3k(pos));
+    c3_c* who_c = u3r_string(who);
+    u3z(who);
+    u3l_log("boot: retrieving keys for sponsor %s\r\n", who_c);
+    free(who_c);
 
     u3_noun oct = u3dc("point:give:dawn", u3k(bok), u3k(pos));
     u3_noun luh = _dawn_eth_rpc(url_c, u3k(oct));
 
-    pon = _dawn_need_unit(u3dc("point:take:dawn", u3k(pos), u3k(luh)),
-                          "boot: failed to retrieve sponsor keys");
-    pon = u3nc(pos, pon);
+    u3_noun son = _dawn_need_unit(u3dc("point:take:dawn", u3k(pos), u3k(luh)),
+                                  "boot: failed to retrieve sponsor keys");
+
+    // append to sponsor chain list
+    //
+    pon = u3nc(u3nc(u3k(pos), son), pon);
     u3z(oct); u3z(luh);
-  }
 
+    // find next sponsor
+    //
+    u3z(ship); u3z(rank);
+    ship = pos;
+    rank = u3do("clan:title", u3k(ship));
+
+    {
+      u3_noun net = _dawn_need_unit(u3k(u3h(u3t(son))),
+                                    "boot: failed to retrieve sponsor");
+      pos = u3k(u3t(u3h(u3t(u3t(u3t(net))))));  //  who.sponsor.u.net.point:azimuth
```

Comment:
Github lost a long review comment I had attached here. I've never seen that before ...

I was just pointing out that `+veri:dawn` internally does this same de-structuring of `+point:azimuth`. I wanted to avoid coupling vere to the internals of that structure, but the I did it makes for a strange API. I'd be inclined to change `+veri:dawn` and always extract the sponsor the same way, just for consistency and clarity.

Reply:
In this case, we'd have to handle the moon and comet cases in C, which I'd like to avoid.  I'll write a `+sponsor:dawn` to extract the sponsor from any ship.